File: src/ts/core/Suit.ts

const SUIT_SIZE = 13; // 13 cards per suit

enum Suit {
  Spades = "spades",
  Hearts = "hearts",
  Clubs = "clubs",
  Diamonds = "diamonds",
}

enum Color {
  Red,
  Black,
}

/**
 * Gets the color of a suite.
 */
 function suitColor(suit: Suit) {
  if (suit === Suit.Spades || suit === Suit.Clubs) {
    return Color.Black;
  }
  return Color.Red;
}

export { SUIT_SIZE, Suit, Color, suitColor }

File: src/suit.rs
use strum::EnumIter;

use crate::macros::ez_display;

#[derive(Copy, Clone, Debug, PartialEq, EnumIter)]
pub enum Suit {
    Spade,
    Diamond,
    Club,
    Heart,
}

#[derive(Copy, Clone, Debug, PartialEq, EnumIter)]
pub enum SuitColor {
    Red,
    Black,
}

impl Suit {
    pub fn color(&self) -> SuitColor {
        match self {
            Suit::Spade | Suit::Club => SuitColor::Black,
            Suit::Diamond | Suit::Heart => SuitColor::Red,
        }
    }
}

ez_display!(Suit);
ez_display!(SuitColor);
